head_first_design_patterns/tbruinem/json/src/include/json_value.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <cstdint>
+
+namespace json {
+
+union JSONValue {
+	int8_t boolean;
+	int64_t bigint;
+	uintptr_t pointer;
```

Comment:
Nice abstraction

Reply:
Because it's a union multiple pointers won't increase the size so I don't see the point in adding this extra abstraction